src/resources/ServiceAccounts/TokenRequestModal/TokenRequestModal.tsx

```diff
@@ -0,0 +1,171 @@
+import { Button, Dialog, MessageStrip } from 'fundamental-react';
+import { useTranslation } from 'react-i18next';
+import { useCallback } from 'react';
+
+import { useGenerateTokenRequest } from './useGenerateTokenRequest';
+import { useDownloadKubeconfigWithToken } from '../useDownloadKubeconfigWithToken';
+import { ResourceForm } from 'shared/ResourceForm';
+import { ComboboxInput } from 'shared/ResourceForm/inputs';
+import { CopiableText } from 'shared/components/CopiableText/CopiableText';
+import { Editor } from 'shared/components/MonacoEditorESM/Editor';
+
+import { useEventListener } from 'hooks/useEventListener';
+import './TokenRequestModal.scss';
+
+const expirationSecondsOptions = [
+  {
+    text: '3600s (1h)',
+    key: 3600,
+  },
+  {
+    text: '21600s (6h)',
+    key: 21600,
+  },
+  {
+    text: '86400s (1d)',
+    key: 86400,
+  },
+  {
+    text: '604800s (7d)',
+    key: 604800,
+  },
+];
+
+const ComboboxInputWithSeconds = ({
+  value,
+  setValue,
+}: {
+  value: number;
+  setValue: (value: number) => void;
+}) => {
+  return (
+    //@ts-ignore
```

Comment:
Because `Single` and the rest of the components aren't written in Typescript. TS says that the props are missing because it thinks all of them are required.